src/types/image.d.ts

```diff
@@ -0,0 +1,12 @@
+interface IImageMedia {
+  [key: string]: string;
+}
+
+export interface IImage {
+  id?: string | number;
+  fallback: string;
+  srcset?: string;
+  media?: IImageMedia;
```

Comment:
suggestion: maybe we should even split this two versions into separate Interfaces? That would it make more clear in the code if a `sources` or `srcset` image will be created.

Reply:
Created two separated intefaces (that extend the base interface)